### PR TITLE
feature/BOT-694-core-support-custom-logic-hooks

### PR DESCRIPTION
**Pull request by ujja**

### CODE REVIEW

**Comment by ujja** [comment]
> The name UPDATE_CONTEXT is not a suitable name, as "context" term is too overloaded in chatbot domain. Lets just name it UPDATE_CUSTOM.
> For updating the watson context variables, it will then look like this:
> 
> ```
> #me
> UPDATE_CUSTOM SET_WATSON_CONTEXT|VarName|VarValue
> ```

done

File: src/scripting/Convo.js [hunk 1]
```diff
@@ -274,7 +274,11 @@ class Convo {
 
           return this.scriptingEvents.setUserInput({ convo: this, convoStep, container, scriptingMemory, meMsg: convoStep })
             .then(() => debug(`${this.header.name}/${convoStep.stepTag}: user says ${JSON.stringify(convoStep, null, 2)}`))
-            .then(() => this.scriptingEvents.onMeStart({ convo: this, convoStep, container, scriptingMemory }))
+            .then(() => {
+              const meMsg = new BotiumMockMessage(convoStep)
```
Comment (codeforequity-at) [inline src/scripting/Convo.js]: I think this should actually happen before line 275, and meMsg should then always be handed over from transcriptStep.actual

File: test/scripting/logichooks/convos/update_custom_me_msg_depth2.convo.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+update_context_me_msg
+updating the me-message.
+
+#me
+HI
+UPDATE_CONTEXT depth1Field|depth2Field|new message
```
Comment (codeforequity-at) [inline test/scripting/logichooks/convos/update_custom_me_msg_depth2.convo.txt]: Please also rename to UPDATE_CUSTOM

Comment (ujja) [reply test/scripting/logichooks/convos/update_custom_me_msg_depth2.convo.txt]: fixed

File: src/scripting/Convo.js (hunk 1 above)
Comment (ujja) [reply src/scripting/Convo.js]: I tried it, but so media is not extracted from convostep. (for example test/scripting/userinputs/mediaInputConvos.spec.js fails)

Comment (codeforequity-at) [reply src/scripting/Convo.js]: hm then maybe the unit test is defect as well. i will take care of it

**Pull request merged**